commit 11ab38d1f1db4daf8150cd887a50ec5d372a3587
Author: Matthew Booth <mbooth@redhat.com>
Date:   Mon Aug 16 14:33:49 2010 +0100

    Allow absolute paths in virt-v2v.conf
    
    This patch allows paths in virt-v2v.conf to be either relative or absolute. If
    relative, they are relative to software-root.
    
    This allows virt-v2v.conf to use files provided by packages independent of
    virt-v2v.

diff --git a/lib/Sys/VirtV2V/Config.pm b/lib/Sys/VirtV2V/Config.pm
index f703152..121e774 100644
--- a/lib/Sys/VirtV2V/Config.pm
+++ b/lib/Sys/VirtV2V/Config.pm
@@ -118,14 +118,16 @@ sub get_transfer_iso
     foreach my $path ($dom->findnodes('/virt-v2v/app/path/text()')) {
         $path = $path->getData();
 
-        # Get the absolute path if iso-root was defined
         my $abs;
-        if (defined($root)) {
-            $abs = File::Spec->catfile($root, $path);
-        } else {
+        if (File::Spec->file_name_is_absolute($path) || !defined($root)) {
             $abs = $path;
         }
 
+        # Make relative paths relative to iso-root if it was defined
+        else {
+            $abs = File::Spec->catfile($root, $path);
+        }
+
         if (-r $abs) {
             $path_args{"$path=$abs"} = 1;
             $paths{$abs} = 1;
